Add warning about closing spans in APM Go SDK Docs (#25852)
DataDog/documentation: content/en/tracing/trace_collection/custom_instrumentation/go/dd-api.md

span.Finish(tracer.WithError(err))
```

> Note: Closing a span that was not started in your code will lead to undefined behaviour.

Author: ```suggestion
Note: Closing a span that was not started in your code can lead to missing data.
```

> Please refer to your specific dd-trace-go [integration][10] documentation to do that.

Author: ```suggestion
Please follow your specific dd-trace-go [integration documentation][10] to do that.
```
